src/key.rs

```diff
@@ -437,6 +437,56 @@ where
     }
 }
 
+pub struct StringDMA<'a> {
+    key: &'a RedisKeyWritable,
+    buffer: &'a mut [u8],
+}
+
+impl<'a> StringDMA<'a> {
+    fn new(key: &'a RedisKeyWritable) -> Result<StringDMA<'a>, RedisError> {
+        let mut length: size_t = 0;
+        let dma = raw::string_dma(key.key_inner, &mut length, raw::KeyMode::WRITE);
+        if dma.is_null() {
+            Err(RedisError::Str("Could not read key"))
+        } else {
+            let buffer = unsafe { std::slice::from_raw_parts_mut(dma.cast::<u8>(), length) };
+            Ok(StringDMA { key, buffer })
+        }
+    }
+
+    pub fn write(&mut self, data: &[u8]) -> Result<&mut Self, RedisError> {
```

Comment:
Why not implement `std::ops::index`and `std::ops::index_mut`?
https://doc.rust-lang.org/std/ops/trait.Index.html
https://doc.rust-lang.org/std/ops/trait.IndexMut.html

Reply:
It's an idea, but it can't be as efficient as a bulk write, I'll add it anyway.

Reply:
Any added Index and IndexMut too